pyro/ops/integrator.py

```diff
@@ -59,9 +59,11 @@ def single_step_velocity_verlet(z, r, potential_fn, step_size, z_grads=None):
 
 
 def _grad(potential_fn, z):
-    z = {k: Variable(v, requires_grad=True) for k, v in z.items()}
     z_keys, z_nodes = zip(*z.items())
+    for node in z_nodes:
```

Comment:
@apaszke, @gchanan - As mentioned in #869, I am setting `requires_grad=True` here right before calling `.grad` and then unsetting it later. If we set this to `True` at the time of tensor declaration, we can just call `.grad` directly here without having to set/unset these flags, but that ends up recording a lot of other tensor operations that we don't need to and is much slower as a result.

Reply:
Yup that looks good